projects/cds/src/merchandising/adapters/strategy/cds-merchandising-strategy.adapter.ts

```diff
@@ -18,16 +18,29 @@ export class CdsMerchandisingStrategyAdapter
 
   loadProductsForStrategy(
     strategyId: string,
-    strategyRequest?: StrategyRequest
+    strategyRequest: StrategyRequest = {}
   ): Observable<StrategyProducts> {
+    let httpHeaders: HttpHeaders = new HttpHeaders();
+    if (
+      strategyRequest &&
+      strategyRequest.headers &&
+      strategyRequest.headers['consent-reference']
+    ) {
+      httpHeaders = new HttpHeaders().set(
+        'consent-reference',
```

Comment:
what's the difference between this header and the one set by consent-reference-interceptor? (sorry I got confused) - just to know

Reply:
Answered above